system/Autoloader/Autoloader.php

```diff
@@ -306,10 +307,28 @@ public function sanitizeFilename(string $filename): string
         // Plus the forward slash for directory separators since this might be a path.
         // http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_278
         // Modified to allow backslash and colons for on Windows machines.
-        $filename = preg_replace('/[^0-9\p{L}\s\/\-\_\.\:\\\\]/u', '', $filename);
+        $result = preg_match_all('/[^0-9\p{L}\s\/\-_.:\\\\]/u', $filename, $matches);
+
+        if ($result > 0) {
+            $chars = implode('', $matches[0]);
+
+            throw new InvalidArgumentException(
+                'The file path contains special characters "' . $chars
+                . '" that are not allowed: "' . $filename . '"'
+            );
+        }
+        if ($result === false) {
+            throw new RuntimeException(preg_last_error_msg() . ' filename: "' . $filename . '"');
```

Comment:
When would this occur? `preg_match_all()`:

> Returns the number of full pattern matches (which might be zero), or false on failure.

Reply:
When regex error occurs.

Reply:
But *how* can it occur? The regex pattern is fixed so we know it won't cause an error, and this function should work with any string input which is enforced by the parameter `string $filename`. I'm fine with extra sanity checks but if we can't actually create the scenario it will lower our code coverage.

Reply:
Added test.